### PR TITLE
Update to 0.9.43

### PR DESCRIPTION
**Pull request by JoshOrndorff**
Update Substrate dependency to `polkadot-v0.9.43`.

### CODE REVIEW

**Review by pmikolajczyk41 — APPROVED** [review]
lgtm but I don't know what is the problem with CI test :|

**Comment by pmikolajczyk41** [comment]
oh, I see: `error: unexpected argument '--ws-port' found` - it must have changed recently

File: docker/academy-chain-compose.yml [hunk 1]
```diff
@@ -16,7 +16,6 @@ services:
       - NODE_KEY_FILE=/data/$NODE01_ACCOUNT_ID/p2p_secret.txt
       - PORT=30333
       - RPC_PORT=9933
-      - WS_PORT=9944
```
Comment (fbielejec) [inline docker/academy-chain-compose.yml:19]: is both the http and ws connectivity now on the same port in this Substarte version? If yes then:

- make sure the first node (the one that exposes and binds ports to the host) uses whatever default ports the tools such as contratcs-ui or polkadot/js wallet expect (this used to be to ws on port 9944)
- remove the port that is not needed from the exposed and from the host mapping

File: node/Cargo.toml [hunk 2]
```diff
@@ -22,32 +22,32 @@ rand = { version = "0.8.5", features = ["small_rng"] }
...
-sp-api = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-blockchain = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-consensus = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-consensus-pow = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-core = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-inherents = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-runtime = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-timestamp = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-transaction-pool = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
+sc-consensus-pow = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.43" }
```
Comment (fbielejec) [inline node/Cargo.toml:30]: NIT: I noticed this before, but strange way to format (left curly no space, right with a space). Maybe a good time to get it consistent.

File: node/sha3pow/Cargo.toml [hunk 3]
```diff
@@ -9,11 +9,11 @@ parity-scale-codec = '3.1.2'
...
-sc-consensus-pow = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sc-client-api = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-api = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-blockchain = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-consensus-pow = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-core = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
-sp-runtime = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
+sc-consensus = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.43" }
+sc-consensus-pow = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.43" }
+sc-client-api = {git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.43" }
```
Comment (fbielejec) [inline node/sha3pow/Cargo.toml:14]: same here, inconsistent brackets formatting

File: node/Cargo.toml (hunk 2 above)
Comment (JoshOrndorff) [reply node/Cargo.toml:30]:
```suggestion
sc-consensus-pow = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.43" }
```

File: node/sha3pow/Cargo.toml (hunk 3 above)
Comment (JoshOrndorff) [reply node/sha3pow/Cargo.toml:14]: Oh yeah, it's on a bunch of them. I never noticed.

File: docker/academy-chain-compose.yml (hunk 1 above)
Comment (JoshOrndorff) [reply docker/academy-chain-compose.yml:19]: https://github.com/paritytech/substrate/pull/13384

Comment (JoshOrndorff) [reply docker/academy-chain-compose.yml:19]: The default is now 9944 for both http and ws. Explained in https://github.com/paritytech/substrate/pull/12663#discussion_r1026563695 and confirmed by my own testing.

File: docker/academy-chain-compose.yml [hunk 4]
```diff
@@ -15,19 +15,15 @@ services:
...
       - BOOT_NODES=/ip4/127.0.0.1/tcp/30333/p2p/$NODE01_PEER_ID
       - PUBLIC_ADDR=/ip4/127.0.0.1/tcp/30333
     volumes:
       - $BASE_PATH/:/data/
     expose:
-      - 30333
-      - 9933
       - 9944
     ports:
-      - 0.0.0.0:30333:30333
```
Comment (fbielejec) [inline docker/academy-chain-compose.yml:30]: that's a bit too much - if you remove the p2p port mapping nodes might not be able to talk

File: .github/scripts/interact-with-contract.sh [hunk 5]
```diff
@@ -14,7 +14,7 @@ $CC cargo contract new --target-dir /sources flipper
 $CC cargo contract build --release --manifest-path=/sources/flipper/Cargo.toml
 
 # Upload and instantiate code
-ADDRESS=$($CC cargo contract instantiate --suri //Alice --skip-confirm --args false -x --output-json /sources/flipper/target/ink/flipper.wasm | jq -jr '.contract')
+ADDRESS=$($CC cargo contract instantiate --suri //Alice --url ws://localhost:9933 --skip-confirm --args false -x --output-json /sources/flipper/target/ink/flipper.wasm | jq -jr '.contract')
```
Comment (fbielejec) [inline .github/scripts/interact-with-contract.sh:17]: should be 9944 I guess, since that is what you're exposing?

**Comment by fbielejec** [comment]
> lgtm but I don't know what is the problem with CI test :|

CI test fails because of the port change - the curl requests that ping the node should now be using the rpc port 9944

File: .github/scripts/interact-with-contract.sh (hunk 5 above)
Comment (JoshOrndorff) [reply .github/scripts/interact-with-contract.sh:17]:
```suggestion
ADDRESS=$($CC cargo contract instantiate --suri //Alice --url ws://localhost:9944 --skip-confirm --args false -x --output-json /sources/flipper/target/ink/flipper.wasm | jq -jr '.contract')
```

**Review by fbielejec — APPROVED** [review]
LGTM, nice!

**Comment by JoshOrndorff** [comment]
Oops, forgot about this one. I've already updated it quite a bit more. I did break the ci big time though. I'll come back to that.

**Pull request closed without merging**